File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FollowRequestsResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.entity.Account;
import mastodon4j.entity.share.Response;

/**
 * @author hecateball
 */
public interface FollowRequestsResource {

    /**
     * Fetching a list of follow requests.
     *
     * @return an array of Accounts which have requested to follow the authenticated user
     */
    public Response<Account[]> getFollowRequests();

    /**
     * Authorizing follow requests.
     *
     * @param id
     */
    public Response<Void> authorizeFollowRequest(String id);

    /**
     * Rejecting follow requests.
     *
     * @param id
     */
    public Response<Void> rejectFollowRequest(String id);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FollowRequestsResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.followrequests.FollowRequestsAuthorizeFollowRequestRequest
import work.socialhub.kmastodon.api.request.followrequests.FollowRequestsRejectFollowRequestRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.ResponseUnit
import work.socialhub.kmastodon.api.response.followrequests.FollowRequestsFollowRequestsResponse
import kotlin.js.JsExport

@JsExport
interface FollowRequestsResource {

    /**
     * Fetching a list of follow requests.
     */
    suspend fun followRequests(
    ): Response<Array<FollowRequestsFollowRequestsResponse>>

    @JsExport.Ignore
    fun followRequestsBlocking(
    ): Response<Array<FollowRequestsFollowRequestsResponse>>

    /**
     * Authorizing follow requests.
     */
    suspend fun authorizeFollowRequest(
        request: FollowRequestsAuthorizeFollowRequestRequest
    ): ResponseUnit

    @JsExport.Ignore
    fun authorizeFollowRequestBlocking(
        request: FollowRequestsAuthorizeFollowRequestRequest
    ): ResponseUnit

    /**
     * Rejecting follow requests.
     */
    suspend fun rejectFollowRequest(
        request: FollowRequestsRejectFollowRequestRequest
    ): ResponseUnit

    @JsExport.Ignore
    fun rejectFollowRequestBlocking(
        request: FollowRequestsRejectFollowRequestRequest
    ): ResponseUnit
}
